v0/destinations/pipedrive/transform.js

```diff
@@ -0,0 +1,416 @@
+/* eslint-disable prefer-const */
+/* eslint-disable prettier/prettier */
+/* eslint-disable camelcase */
+const get = require("get-value");
+const set = require("set-value");
+const { EventType } = require("../../../constants");
+const {
+  constructPayload,
+  extractCustomFields,
+  defaultRequestConfig,
+  removeUndefinedAndNullValues,
+  defaultPostRequestConfig,
+  defaultPutRequestConfig,
+  getValueFromMessage,
+} = require("../../util");
+const {
+  MAPPING_CONFIG,
+  CONFIG_CATEGORIES,
+  PERSONS_ENDPOINT,
+  PIPEDRIVE_IDENTIFY_EXCLUSION,
+  PIPEDRIVE_GROUP_EXCLUSION,
+  PIPEDRIVE_PRODUCT_EXCLUSION,
+  PIPEDRIVE_TRACK_EXCLUSION,
+  getMergeEndpoint,
+  LEADS_ENDPOINT,
+  PRODUCTS_ENDPOINT,
+  PRODUCT_EVENTS
+} = require("./config");
+const {
+  createNewOrganisation,
+  searchPersonByCustomId,
+  searchOrganisationByCustomId,
+  getFieldValueOrThrowError,
+  updateOrganisationTraits,
+  updatePerson,
+  renameCustomFields
+} = require("./util");
+
+/** 
+ * Config structure so far
+ * {
+ *    "api_token": "asfaaf31423csaf",
+ *    "groupIdToken": "sfhijsfh",
+ *    "userIdToken": "nzoicb88714sCV"
+ * }
+ * 
+ * New potential Config structure
+ * shall contain all custom field id vs api_key mappings
+ * {
+ *    "api_token": "asfaaf31423csaf",
+ *    "r_group_id": "sfhijsfh",
+ *    "r_user_id": "nzoicb88714sCV",
+ *    "customFieldsMap": {
+ *        "person": {
+ *          "relation_status": "relstatusapi_key",
+ *          "district": "distapi_key",
+ *          ...
+ *        },
+ *        "product": {...},
+ *        "leads": {...},
+ *        "organization": {...},
+ *        "track": {...}
+ *     }
+ * }
+ * 
+*/
+const identifyResponseBuilder = async (message, category, destination) => {
+  // name is required field. If name is not present, construct payload will
+  // throw error
+  let payload = constructPayload(message, MAPPING_CONFIG[category.name]);
+
+  payload = extractCustomFields(
+    message,
+    payload,
+    ["traits", "context.traits"],
+    PIPEDRIVE_IDENTIFY_EXCLUSION
+  );
+
+  payload = renameCustomFields(payload, destination.Config.fieldsMap, "person");
+
+  const userIdValue = getFieldValueOrThrowError(
```

Comment:
lets do this beofre creating the payload. as no use of creating payload if user id is not present.

Reply:
right. will make similar corrections in other places too, if any.